Mathlib/LinearAlgebra/Basic.lean

```diff
@@ -546,10 +547,12 @@ def smulOfUnit (a : Rˣ) : M ≃ₗ[R] M :=
 /-- A linear isomorphism between the domains and codomains of two spaces of linear maps gives a
 linear isomorphism between the two function spaces. -/
 def arrowCongr {R M₁ M₂ M₂₁ M₂₂ : Sort _} [CommSemiring R] [AddCommMonoid M₁] [AddCommMonoid M₂]
-    [AddCommMonoid M₂₁] [AddCommMonoid M₂₂] [Module R M₁] [Module R M₂] [Module R M₂₁]
-    [Module R M₂₂] (e₁ : M₁ ≃ₗ[R] M₂) (e₂ : M₂₁ ≃ₗ[R] M₂₂) : (M₁ →ₗ[R] M₂₁) ≃ₗ[R] M₂ →ₗ[R] M₂₂ where
-  toFun := fun f : M₁ →ₗ[R] M₂₁ => (e₂ : M₂₁ →ₗ[R] M₂₂).comp <| f.comp (e₁.symm : M₂ →ₗ[R] M₁)
-  invFun f := (e₂.symm : M₂₂ →ₗ[R] M₂₁).comp <| f.comp (e₁ : M₁ →ₗ[R] M₂)
+    [AddCommMonoid M₂₁] [AddCommMonoid M₂₂] [Module R₁ M₁] [Module R₁ M₂] [Module R₁ M₂₁]
+    [Module R₁ M₂₂] [Module R M₂₁] [Module R M₂₂] [SMulCommClass R₁ R M₂₁] [SMulCommClass R₁ R M₂₂]
+    (e₁ : M₁ ≃ₗ[R₁] M₂) (e₂ : M₂₁ ≃ₗ[R₁] M₂₂) (h : ∀ (c : R) (x), e₂ (c • x) = c • e₂ x) :
```

Comment:
Is it acceptable in your use case to avoid the need for `h` by instead requiring that R` and `R₁` form a tower?

Reply:
Concretely: what's the specialization that you care about here? I think this is treading a bit close to "we need maps that are linear over multiple rings simultaneously", which is a can of worms we might want to spend more time designing an opener for.

Reply:
@eric-wieser it all stems from https://github.com/leanprover-community/mathlib4/pull/9334#pullrequestreview-1880856848 -

Reply:
I think I'd be fine with all the rings being equal, but you wanted more for `LinearMap.toMatrix₂Aux` (https://github.com/leanprover-community/mathlib4/pull/9334#discussion_r1490154631) so I assume that applies to `LinearMap.toMatrix₂` as well?